src/cache/chrome-storage-async-map.ts

```diff
@@ -18,7 +18,36 @@ export default class ChromeStorageAsyncMap<T> implements AsyncMap<string, T> {
     return await this.storage.get(null);
   }
 
-  async set(key: string, value: T) {
-    await this.storage.set({ [key]: value });
+  async set(key: string, value: T): Promise<void> {
+    try {
+      await new Promise<void>((resolve) => {
+        // Set up listener for this specific write
+        const listener = (changes: { [key: string]: chrome.storage.StorageChange }) => {
+          if (changes[key]) {
+            chrome.storage.onChanged.removeListener(listener);
+            resolve();
+          }
+        };
+
+        chrome.storage.onChanged.addListener(listener);
+
+        // Set a timeout in case the change event never fires
+        const timeout = setTimeout(() => {
+          chrome.storage.onChanged.removeListener(listener);
+          applicationLogger.warn('Chrome storage write timeout for key:', key);
+          resolve();
+        }, 1000);
```

Comment:
One second is probably too long, but we also never expect to end up in this situation of needing the timeout. It's a precaution